dpdata/customsystem.py

```diff
@@ -0,0 +1,436 @@
+        ret+="\n"+"  ".join(map(str,self.get_atom_numbs()))
+        if (len(self.mols) > 0):
+            ret+="\nMolecule List:\n"
+            for item in self.mols:
+                mol = item["mol"]
+                at = item["at"]
+                at_is_cartesian = item["at_is_cartesian"]
+                ret+=mol.__str__()
+                if at_is_cartesian:
+                    ret+="\n"+"@ cartesian coordinate:"
+                    ret+="\n"+"  ".join(map(str,at))
+                    ret+="\n"
+                else:
+                    ret+="\n"+"@ direct coordinate:"
+                    ret+="\n"+"  ".join(map(str,at))
+                    ret+="\n"
+        return ret
+
+    def __getitem__(self, key):
+        """Returns proerty stored in System by key or by idx"""
+        if isinstance(key, (int, slice)):
+            return self.sub_system(key)
+        data_hybrid = deepcopy(self.data)
+        for item in self.mols:
+            mol = item["mol"]
+            at = item["at"]
+            at_is_cartesian = item["at_is_cartesian"]
+            data_hybrid = self._merge_data(data_hybrid, mol, at, at_is_cartesian)
+        return data_hybrid[key]
+
+    def get_nmols(self) :
+        """Returns number of frames in the system"""
+        return len(self.mols)
+
+
+    def append(self, system) :
+        """
+        Append a system to this system
+
+        Parameters
+        ----------
+        system : System
+            The system to append
+        """
+        if not len(system.data['atom_numbs']):
+            # skip if the system to append is non-converged
+            return False
+        elif not len(self.data['atom_numbs']):
+            # this system is non-converged but the system to append is converged
+            self.data = system.data
+            return False
+        if system.uniq_formula != self.uniq_formula:
+            raise RuntimeError('systems with inconsistent formula could not be append: %s v.s. %s' % (self.uniq_formula, system.uniq_formula))
+        if system.data['atom_names'] != self.data['atom_names']:
+            # allow to append a system with different atom_names order
+            system.sort_atom_names()
+            self.sort_atom_names()
+        if (system.data['atom_types'] != self.data['atom_types']).any():
+            # allow to append a system with different atom_types order
+            system.sort_atom_types()
+            self.sort_atom_types()
+        for ii in ['atom_numbs', 'atom_names'] :
+            assert(system.data[ii] == self.data[ii])
+        for ii in ['atom_types','orig'] :
+            eq = [v1==v2 for v1,v2 in zip(system.data[ii], self.data[ii])]
+            assert(all(eq))
+        for ii in ['coords', 'cells'] :
+            self.data[ii] = np.concatenate((self.data[ii], system.data[ii]), axis = 0)
+        if self.nopbc and not system.nopbc:
+            # appended system uses PBC, cancel nopbc
+            self.data['nopbc'] = False
```

Comment:
Wring duplicated codes should be avoid. One can use
```suggestion
        super().append(system)
```
to call the parent's method.

Same for other methods.